File: src/test/java/no/nav/syfo/repository/CarDAOTest.java
package no.nav.syfo.repository;

import no.nav.syfo.LocalApplication;
import no.nav.syfo.domain.Car;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.context.junit4.SpringRunner;

import javax.inject.Inject;
import java.time.LocalDate;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = LocalApplication.class)
@DirtiesContext
public class CarDAOTest {

    @Inject
    private JdbcTemplate jdbcTemplate;

    @Inject
    private CarDAO carDAO;

    @Before
    public void cleanup() {
        jdbcTemplate.update("DELETE FROM CAR");
    }

    @Test
    public void hentCars() {
        jdbcTemplate.update("INSERT INTO CAR VALUES('id', 'blue', 'Lada', date '2018-05-05')");

        List<Car> cars = carDAO.hentCars();

    }

    @Test
    public void lagreCar() {
        Car car = Car.builder()
                .color("red")
                .brand("Mercedes")
                .bought(LocalDate.of(2018, 6, 27))
                .build();
        carDAO.lagreCar(car);

        List<Car> cars = jdbcTemplate.query("SELECT * FROM CAR", CarDAO.getInnsendingRowMapper());
        assertThat(cars.size()).isEqualTo(1);
        assertThat(cars.get(0).getColor()).isEqualTo("red");
    }
}

File: src/test/kotlin/no/nav/syfo/repository/CarDAOTest.kt
package no.nav.syfo.repository

import no.nav.syfo.LocalApplication
import no.nav.syfo.domain.Car
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.test.annotation.DirtiesContext
import org.springframework.test.context.junit4.SpringRunner
import java.time.LocalDate
import javax.inject.Inject

@RunWith(SpringRunner::class)
@SpringBootTest(classes = [LocalApplication::class])
@DirtiesContext
class CarDAOTest {

    @Inject
    private val jdbcTemplate: JdbcTemplate? = null

    @Inject
    private val carDAO: CarDAO? = null

    @Before
    fun cleanup() {
        jdbcTemplate!!.update("DELETE FROM CAR")
    }

    @Test
    fun hentCars() {
        jdbcTemplate!!.update("INSERT INTO CAR VALUES('id', 'blue', 'Lada', date '2018-05-05')")

        val cars = carDAO!!.hentCars()
    }

    @Test
    fun lagreCar() {
        val car = Car(
            color = "red",
            brand = "Mercedes",
            bought = LocalDate.of(2018, 6, 27)
        )
        carDAO!!.lagreCar(car)

        val cars = jdbcTemplate!!.query("SELECT * FROM CAR", carDAO.innsendingRowMapper)
        assertThat(cars).hasSize(1)
        assertThat(cars.first().color).isEqualTo("red")
    }
}